lib/routes/run.js

```diff
@@ -1,20 +1,63 @@
 'use strict';
 var chalk = require('chalk');
+var findUp = require('find-up');
+var inquirer = require('inquirer');
 var namespaceToName = require('yeoman-environment').namespaceToName;
+var path = require('path');
+var userHome = require('user-home');
+
+// Checks if .yo-rc.json exists in parent directory
+var filePath = findUp.sync('.yo-rc.json');
+if (filePath !== null) {
+  filePath = path.dirname(filePath);
+}
+
+function scaffoldInto(app, name) {
+  var baseName = namespaceToName(name);
+  inquirer.prompt([{
+    name: 'scaffold',
+    type: 'confirm',
+    message: 'Are you sure you want to scaffold in ' + filePath +
+      ' ?'
+  }], function (answer) {
+    if (answer.scaffold) {
+      console.log(
+        chalk.yellow('\nMake sure you are in the directory you want to scaffold into.') +
+        chalk.dim('\nThis generator can also be run with: ') +
+        chalk.blue('yo ' + baseName + '\n')
+      );
+
+      // save the generator run count
+      var generatorRunCount = app.conf.get('generatorRunCount');
+      generatorRunCount[baseName] = generatorRunCount[baseName] + 1 || 1;
+      app.conf.set('generatorRunCount', generatorRunCount);
+      app.env.run(name);
+    } else {
+      return;
+    }
+  });
+}
 
 module.exports = function (app, name) {
   var baseName = namespaceToName(name);
   app.insight.track('yoyo', 'run', baseName);
 
-  console.log(
-    chalk.yellow('\nMake sure you are in the directory you want to scaffold into.') +
-    chalk.dim('\nThis generator can also be run with: ') +
-    chalk.blue('yo ' + baseName + '\n')
-  );
+  if (filePath !== null && filePath !== process.cwd()) {
+    scaffoldInto(app, name);
+  } else if (process.cwd() === userHome) {
+    filePath = userHome;
```

Comment:
setting this doesnt seem to do anything since filePath isn't used after this. This if block can be combined with the one above it then.